UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -578,18 +571,15 @@ void SendInfoToBrain()
             m_Info.maxStepReached = m_MaxStepReached;
             m_Info.id = m_Id;
 
-            m_Brain.RequestDecision(this);
+            m_Brain.RequestDecision(m_Info, sensors, UpdateAgentAction);
 
             if (m_Recorder != null && m_Recorder.record && Application.isEditor)
             {
                 // This is a bit of a hack - if we're in inference mode, observations won't be generated
                 // But we need these to be generated for the recorder. So generate them here.
-                if (m_Info.observations.Count == 0)
-                {
-                    GenerateSensorData();
-                }
+                var observations = GenerateSensorData(sensors, m_VectorSensorBuffer, m_WriteAdapter);
```

Comment:
We're allocating m_VectorSensorBuffer for every agent but (I think) only using it here now.

Reply:
I allocate only when recording now.